src/signers.rs

```diff
@@ -0,0 +1,266 @@
+            || psbt.inputs[input_index].final_script_witness.is_some()
+        {
+            return Ok(());
+        }
+
+        let child_matches = psbt.inputs[input_index].bip32_derivation.iter().find(
+            |(_, &(fingerprint, ref path))| {
+                if fingerprint != signer_fingerprint {
+                    return false;
+                }
+                if self.path.len() > path.len() {
+                    return false;
+                }
+                for (i, child_n) in self.path.into_iter().enumerate() {
+                    if path[i] != *child_n {
+                        return false;
+                    }
+                }
+                true
+            },
+        );
+
+        let (public_key, full_path) = match child_matches {
+            Some((pk, (_, full_path))) => (pk, full_path.clone()),
+            None => return Ok(()),
+        };
+
+        let deriv_path = DerivationPath::from(
+            full_path
+                .into_iter()
+                .cloned()
+                .skip(self.path.len())
+                .collect::<Vec<ChildNumber>>(),
+        );
+        let derived_key = xkey.derive_priv(secp, &deriv_path).unwrap();
+
+        if &PublicKey::new(secp256k1::PublicKey::from_secret_key(
+            secp,
+            &derived_key.private_key,
+        )) != public_key
+        {
+            Err(SignerError::InvalidKey)
+        } else {
+            PrivateKey::new(derived_key.private_key, Network::Bitcoin).sign(
+                psbt,
+                Some(input_index),
+                secp,
+            )
+        }
+    }
+
+    fn sign_whole_tx(&self) -> bool {
+        false
+    }
+
+    fn id(&self, secp: &Secp256k1<All>) -> SignerId {
+        SignerId::from(self.parent_xkey.fingerprint(secp))
+    }
+}
+
+#[derive(Debug)]
+pub struct PwSeedSigner {
+    /// The derivation path
+    pub path: bip32::DerivationPath,
+    /// Seed Mnemonic (without passphrase)
+    pub mnemonic: Mnemonic,
+    /// Bitcoin network
+    pub network: Network,
+    /// The expected external wallet descriptor
+    pub correct_descriptor: String,
+}
+
+impl PwSeedSigner {
+    pub fn create_signer(self) -> anyhow::Result<XKeySigner> {
+        let mut passphrase = String::new();
+        eprintln!("Please enter your wallet passphrase: ");
+        let _ = std::io::stdin().read_line(&mut passphrase);
+        let full_seed = self.mnemonic.to_seed(passphrase);
+
+        let xkey: ExtendedKey = full_seed.into_extended_key().unwrap();
+        let xpriv = xkey.into_xprv(self.network).unwrap();
+
+        let temp_wallet = Wallet::new_offline(
+            bdk::template::Bip84(xpriv, bdk::KeychainKind::External),
+            Some(bdk::template::Bip84(xpriv, bdk::KeychainKind::Internal)),
+            self.network,
+            MemoryDatabase::new(),
+        )
+        .context("Initializing wallet with xpriv derived from seed phrase")?;
```

Comment:
This is an interesting problem but I don't think this is a robust way of checking it. You could add a passphrase BIP39 signer to a multisig descriptor. It's not possible with `init` but the config structure allows it and it should just work. I think just save the xpub fingerprint and check that.